mdolab/pyoptsparse
# Optview dash

## Purpose
This pull request is in order to add the OptView tool created with the Dash framework to pyoptsparse. I've made the following changes:
- added a OptView_dash.py file and corresponding assets
- deleted old Dash file (and corresponding assets)
- minor changes to setup.py
	- register `optview_dash` as a new CLI tool
	- added optional dependency set "optview"
- moved the old OptView icon to the assets folder
- updated documentation
- added a test for OptView-Dash (cannot test OptView due to GUI)
- updated Travis
- changed line ending for some files from CRLF to LF

Apologies for the large number of file changes due to line ending, I could revert this and submit as a separate PR if it would be better for reviewing this PR.

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [ ] Breaking change (non-backwards-compatible fix or feature)

## Testing
Please do the following to test
1. Install the package (both in-place and full installation are supported)
2. invoke `optview_dash <file>` on an existing optimization history file. You can use one of the files generated by the tests if you like
3. Verify that OptView-Dash works as expected.

## Checklist
_Put an `x` in the boxes that apply._

- [x] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
